@here/harp-omv-datasource/lib/OmvData.ts

```diff
@@ -576,41 +580,121 @@ export class OmvProtobufDataAdapter implements OmvDataAdapter, OmvVisitor {
             return;
         }
 
-        const geometry: IPolygonGeometry[] = [];
-        const currentPolygon: IPolygonGeometry = { rings: [] };
-        let currentRing: IRing;
-        this.m_geometryCommands.accept(
-            feature.geometry,
-            this.m_tileKey,
-            this.m_geoBox,
-            this.m_layer.extent,
-            {
-                type: "Polygon",
-                visitCommand: command => {
-                    if (isMoveToCommand(command)) {
-                        const geoPoint = new GeoCoordinates(command.latitude, command.longitude);
-                        currentRing = { coordinates: [geoPoint], outlines: [command.outline!] };
-                    } else if (isLineToCommand(command)) {
-                        const geoPoint = new GeoCoordinates(command.latitude, command.longitude);
-                        currentRing.coordinates.push(geoPoint);
-                        currentRing.outlines!.push(command.outline!);
-                    } else if (isClosePathCommand(command)) {
-                        currentPolygon.rings.push(currentRing);
+        const env = createFeatureEnv(this.m_layer, feature, "polygon", storageLevel, this.m_logger);
+        const techniques = this.m_processor.getMatchingTechniques(env);
```

Comment:
For consideration ...

This makes us call `getMatchingTechniques` at least two or even 3 times (here, and then in OMVDecodedTileEmitter ...)
`getMatchingTechniques` is quite expensive because it has to execute several `when` expressions not sure if there is way to pass list of already found techniques.